index.bs

```diff
@@ -130,6 +130,71 @@ much like a data property as possible. Specific guidance in this regard includes
     true. (This will not always be the case, e.g. if a normalization or type conversion step is
     necessary, but should be held as a goal for normal code paths.)
 
+<h3 id="live-vs-static">Consider whether objects should be live or static</h3>
+
+Objects returned from functions, attribute getters, etc.,
+can either be live or static.
+A <dfn>live object</dfn> is one that continues to reflect
+changes made after it was returned to the caller.
+A <dfn>static object</dfn> is one that reflects
+the state at the time it was returned.
+
+Objects that are the way state is mutated are generally live.
+For example, DOM nodes are returned as live objects,
+since they are the API through which attributes are set and other changes
+to the DOM are made.
+They also reflect changes to the DOM made in other ways
+(such as through user interaction with forms).
+Even though the nodes themselves are live,
+they might be returned from a static object,
+such as a static {{NodeList}} that is
+the result of {{ParentNode/querySelectorAll()}}.
+
+Objects that represent a collection that might change over time
+should generally be returned as static objects.
+This is because it is confusing to users of the API when
+a collection changes while being iterated.
+Because of this,
+it is generally considered a mistake that methods like
+{{Document/getElementsByTagName()}} return live objects;
+{{ParentNode/querySelectorAll()}} was made to return static objects
+as a result of this experience.
+
+Note:  It's possible that some of this advice should be reconsidered
+for <a>Iterable</a>, <a>Maplike</a>, and <a>Setlike</a> types
+which can be iterated using iterators that have
+well-defined behavior for mutation that happens during iteration.
+(Is this true for all three?)
```

Comment:
It's certainly well-defined. Both for these and for cases like live NodeLists. I think the issue is that it might be confusing to deal with, regardless.